File: mpush_ios_demo/mpush_ios_demo/MsgToolBox.m
//
//  MsgToolBox.m
//  UserExperienceDemo
//
//  Created by liuzhilong on 15/3/31.
//  Copyright (c) 2015年 alibaba. All rights reserved.
//

#import "MsgToolBox.h"

@implementation MsgToolBox

+ (void)showAlert:(NSString *)title content:(NSString *)content {
    // 保证在主线程上执行
    if ([NSThread isMainThread]) {
        [self show:title content:content];
    } else {
        dispatch_async(dispatch_get_main_queue(), ^{
            [self show:title content:content];
        });
    }
}

+ (void)show:(NSString *)title content:(NSString *)content {
    UIAlertController *alert = [UIAlertController alertControllerWithTitle:title message:content preferredStyle:UIAlertControllerStyleAlert];
    UIAlertAction *cancleAction = [UIAlertAction actionWithTitle:@"已阅" style:UIAlertActionStyleCancel handler:nil];
    [alert addAction:cancleAction];
    [[self getCurrentViewController] presentViewController:alert animated:YES completion:nil];
}

+ (UIViewController *)getCurrentViewController {
    UIViewController *rootViewController = [UIApplication sharedApplication].keyWindow.rootViewController;
    return [self getVisibleViewControllerFrom:rootViewController];
}

+ (UIViewController *)getVisibleViewControllerFrom:(UIViewController *)vc {
    if ([vc isKindOfClass:[UINavigationController class]]) {
        return [self getVisibleViewControllerFrom:((UINavigationController *)vc).visibleViewController];
    } else if ([vc isKindOfClass:[UITabBarController class]]) {
        return [self getVisibleViewControllerFrom:((UITabBarController *)vc).selectedViewController];
    } else if (vc.presentedViewController) {
        return [self getVisibleViewControllerFrom:vc.presentedViewController];
    } else {
        return vc;
    }
}

@end

File: mpush_swift_demo/mpush_swift_demo/MsgToolBox.swift
//
//  MsgToolBox.swift
//  mpush_swift_demo
//
//  Created by fuyuan.lfy on 16/7/26.
//  Copyright © 2016年 alibaba.qa. All rights reserved.
//

import UIKit

class MsgToolBox: NSObject {
    static func showAlert(title: String, content: String) {
        // 保证在主线程上执行
        if NSThread.isMainThread() {
            let alertView = UIAlertView.init(title: title, message: content, delegate: nil, cancelButtonTitle: "已阅")
            alertView.show()
        } else {
            dispatch_async(dispatch_get_main_queue(), {
                let alertView = UIAlertView.init(title: title, message: content, delegate: nil, cancelButtonTitle: "已阅")
                alertView.show()
            })
        }
    }
}
